cosmos/dbt/graph.py

```diff
@@ -122,24 +87,47 @@ def load_via_dbt_ls(self):
         * self.filtered_nodes
         """
         logger.info("Trying to parse the dbt project using dbt ls...")
-        command = [self.dbt_cmd, "ls", "--output", "json", "--profiles-dir", self.project.dir]
-        if self.exclude:
-            command.extend(["--exclude", *self.exclude])
-        if self.select:
-            command.extend(["--select", *self.select])
-        logger.info(f"Running command: {command}")
-        try:
+
+        command = [
+            str(self.execution_config.dbt_executable_path),
+            "ls",
+            "--output",
+            "json",
+        ]
+
+        if self.render_config.exclude:
+            command.extend(["--exclude", *self.render_config.exclude])
+
+        if self.render_config.select:
+            command.extend(["--select", *self.render_config.select])
+
+        with self.profile_config.ensure_profile() as (
+            profile_path,
+            env_vars,
+        ):
+            command.extend(["--profiles-dir", str(profile_path.parent)])
+            command.extend(["--profile", self.profile_config.profile_name])
+            command.extend(["--target", self.profile_config.target_name])
+            env = {
+                **os.environ,
+                **env_vars,
+            }
+
+            logger.info("Running command `%s`", command)
             process = Popen(
-                command, stdout=PIPE, stderr=PIPE, cwd=self.project.dir, universal_newlines=True, env=os.environ
+                command,
+                stdout=PIPE,
+                stderr=PIPE,
+                cwd=self.project_config.dbt_project_path,
+                env=env,
+                universal_newlines=True,
             )
-        except FileNotFoundError as exception:
```

Comment:
This exception matters. It happens when the `dbt` binary path is not found in the system. We should not delete it.

Reply:
what do you think about adding an explicit check that the `dbt` binary is found? it really wasn't clear to me what this exception was supposed to be doing